src/kubernetes_cleanup_pods_by_name/kubernetes_cleanup_pods_by_name/pods_by_name.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime, timedelta
+
+from lib.kctl import Kctl
+from lib.logging import logging
+from kubernetes_cleanup_pods_by_name.config import config
+
+def cleanup_pods_by_name():
+  """Cleanup pods by name older than NHOURS ago"""
+
+  logging.info(
+    "Cleaning up pods in {0} namespace " \
+    "that are older than {1} hours " \
+    "and contain '{2}' in their name"
+    .format(config.namespace, config.nhours, config.name)
+  )
+
+  kctl = kctl_client()
+  pods = kctl.get_pods(config.namespace)
+
+  for pod in pods:
+    pod_name = pod['metadata']['name']
+    if config.name not in pod_name:
+      logging.debug(
+        f"Skipping pod {pod_name} because it does not contain '{config.name}' in its name"
+      )
+      continue
+
+    if 'startTime' not in pod['status']:
+      logging.debug(
+        f"Skipping pod {pod_name} because it does not have a startTime"
+      )
+      continue
+
+    pod_start_time = datetime.strptime(pod['status']['startTime'], "%Y-%m-%dT%H:%M:%SZ")
+
+    if pod_start_time < datetime.utcnow() - timedelta(hours=config.nhours):
+      if config.force:
+        kctl.delete_pod(config.namespace, pod_name)
+        logging.info(
+          f"Deleted pod {pod_name}"
+        )
+      else:
+        logging.info(
+          f"Would have deleted pod {pod_name}"
+        )
+
+  logging.info("Done.")
+
+def kctl_client():
```

Comment:
Wondering if this could be extracted into a shared module or something 🤔

Reply:
oh yes. [done](https://github.com/artsy/opstools/pull/29/commits/051de63310239d0adf02703ffa3c7a77f0faa1b7).i will do the same for other scripts at a later date.